src/test/java/org/opensearch/security/auditlog/compliance/RestApiComplianceAuditlogTest.java

```diff
@@ -73,27 +78,28 @@ public void testRestApiRolesDisabled() throws Exception {
             .build();
 
         setup(additionalSettings);
-        TestAuditlogImpl.clear();
+        Thread.sleep(1000);
         String body = "{ \"password\":\"some new password\",\"backend_roles\":[\"role1\",\"role2\"] }";
 
         rh.enableHTTPClientSSL = true;
         rh.trustHTTPServerCertificate = true;
         rh.sendAdminCertificate = true;
         rh.keystore = "kirk-keystore.jks";
 
-        HttpResponse response = rh.executePutRequest("_opendistro/_security/api/internalusers/compuser?pretty", body);
-        Thread.sleep(1500);
-        Assert.assertEquals(HttpStatus.SC_CREATED, response.getStatusCode());
-        Assert.assertTrue(TestAuditlogImpl.messages.size() + "", TestAuditlogImpl.messages.size() == 1);
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("audit_request_effective_user"));
-        Assert.assertFalse(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_READ"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_WRITE"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("UPDATE"));
-        validateMsgs(TestAuditlogImpl.messages);
+        final List<AuditMessage> messages = TestAuditlogImpl.doThenWaitForMessages(() -> {
+            HttpResponse response = rh.executePutRequest("_opendistro/_security/api/internalusers/compuser?pretty", body);
+            Assert.assertEquals(HttpStatus.SC_CREATED, response.getStatusCode());
+            try {
+                Thread.sleep(1500);
+            } catch (InterruptedException e) {
+                fail("Received unexpected exception");
+            }
+            Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_WRITE"));
```

Comment:
When you translated between the old and the new version you removed a number of assertions, was there a reason you removed them?

Reply:
Yes, because the assertions were only applicable to startup messages. I modified the tests to specifically look for the messages generated from the API call made in the test itself. `testAutoInit` is a special test where its looking for the messages generated on startup and not the messages from any particular API call.